src/containers/booking-confirmation/DevconCashbackReward.tsx

```diff
@@ -153,14 +160,29 @@ export const DevconCashbackReward = () => {
 
     window.negotiator.on(
       'token-proof',
-      (proof: { error: string; data: { proof: string }; issuer: string }) => {
+      (proof: { error: string|Error; data: { proof: string }; issuer: string }) => {
         if (proof.error) {
-          showError(AUTH_ERROR_STR, proof.error);
+          showError(AUTH_ERROR_STR, proof.error instanceof Error ? proof.error.message: proof.error);
           return;
         }
         sendCashbackRequest(proof.data);
       }
     );
+  }
+
+  useEffect(() => {
```

Comment:
In some cases, it will be possible that this script adds the `script` element multiple times.
I recommend adding an `id` parameter to the `script` (`script.id = 'negotiator'`) and check if this `id` exist in the DOM before adding. This way checking of conditions may look like:

```typescript
if (window.negotiator instanceof Client || document.getElementById('negotiator')) {
  return;
}
```